src/indexer/segment_writer.rs

```diff
@@ -157,15 +155,17 @@ impl SegmentWriter {
             + self.segment_serializer.mem_usage()
     }
 
-    fn index_document(&mut self, doc: &Document) -> crate::Result<()> {
+    fn index_document<D: DocumentAccess>(&mut self, doc: &D) -> crate::Result<()> {
         let doc_id = self.max_doc;
+        // TODO: This should be removed...
```

Comment:
Why should this be removed?
There was a reason for the grouping, but I don't remember

Reply:
My original thought was this seems to be a bit of optimisation around the `Document` type storing it's multi-value fields individually, so my original thinking was this could just get removed and we simply go "If you actually give an array iterator for the field then you can probably a bit more efficient and not need to sort and group and do those small allocations" That being said, having used this PR for a bit, I've found myself not really using the array iterator that much just due to how lnx processes documents.